sqlmesh/core/model/definition.py

```diff
@@ -1411,6 +1421,10 @@ class ExternalModel(_Model):
     def is_breaking_change(self, previous: Model) -> t.Optional[bool]:
         if not isinstance(previous, ExternalModel):
             return None
+        # If the default catalog has changed between models then we want to consider that breaking since
+        # that means we could be pointing to a different source.
+        if previous.fqn != self.fqn:
```

Comment:
can you move this to common breaking changes (dry this up) ? i don't like how this is repeated in every model